pkg/commonvpc/testfunc/common_test_functionality.go

```diff
@@ -386,94 +316,39 @@ var formatsAvoidComparisonAndGeneration = map[vpcmodel.OutFormat]bool{
 	}
-}
-
-///////////////////////////////////////////////////////////////////////////////////////////
-// lint:
-//////////////////////////////////////////////////////////////////////////////////////////////
-
-const lintOut = "lint_out"
-
-func RunLintTest(tt *VpcGeneralTest, t *testing.T, rc ResourcesContainer) {
-	// all tests in lint mode
-	// output use case is not significant here, but being used so that lint test can rely on existing mechanism
-	tt.UseCases = []vpcmodel.OutputUseCase{vpcmodel.AllEndpoints}
-	// init test - set the input/output file names according to test name
-	tt.InitTest()
-
-	// get vpcConfigs obj from parsing + analyzing input config file
-	vpcConfigs := GetVPCConfigs(t, tt, true, rc)
-
-	// generate actual output for all use cases specified for this test
-	err := runLintTestPerUseCase(t, tt, vpcConfigs.Configs(), lintOut)
-	require.Equal(t, tt.ErrPerUseCase[vpcmodel.AllEndpoints], err, "comparing lint actual err to expected err")
 	for uc, outFile := range tt.ActualOutput {
-		fmt.Printf("lint test %s use-case %d - generated output file: %s\n", tt.Name, uc, outFile)
-	}
-}
-
-// runExplainTestPerUseCase executes lint for the required use case and compares/generates the output
-func runLintTestPerUseCase(t *testing.T,
-	tt *VpcGeneralTest,
-	cConfigs map[string]*vpcmodel.VPCConfig,
-	outDir string) error {
-	// output use case is not significant here, but being used so that lint test can rely on existing mechanism
-	initLintTestFileNames(tt, outDir)
-	_, actualOutput, _ := linter.LinterExecute(cConfigs, tt.PrintAllLints, tt.Enable, tt.Disable)
-	if err := CompareOrRegenerateOutputPerTest(t, tt.Mode, actualOutput, lintOut, tt, vpcmodel.AllEndpoints); err != nil {
-		return err
+		fmt.Printf("test %s use-case %d - generated output file: %s\n", tt.Name, uc, outFile)
 	}
-	return nil
-}
-
-func initLintTestFileNames(tt *VpcGeneralTest, testDir string) {
-	expectedFileName, actualFileName := getLintTestFileName(tt.Name)
-	// output use case is not significant here, but being used so that lint test can rely on existing mechanism
-	tt.ActualOutput[vpcmodel.AllEndpoints] = filepath.Join(GetTestsDirOut(testDir), actualFileName)
-	tt.ExpectedOutput[vpcmodel.AllEndpoints] = filepath.Join(GetTestsDirOut(testDir), expectedFileName)
 }
 
-// getLintTestFileName returns expected file name and actual file name, for the relevant use case
-func getLintTestFileName(testName string) (expectedFileName, actualFileName string) {
-	res := testName + "_Lint"
-	expectedFileName = res
-	actualFileName = ActualOutFilePrefix + res
-	return expectedFileName, actualFileName
+func (tt *VpcTestCommon) TestCommonSingleTest(t *testing.T, mode testMode, rc commonvpc.ResourcesContainer, testDir,
+	testName string, grouping, noLbAbstract bool) {
+	tt.Name = testName
+	tt.setMode(mode)
+	t.Run(tt.Name, func(t *testing.T) {
+		t.Parallel()
+		tt.runSingleCommonTest(t, testDir, rc, grouping, noLbAbstract, nil)
+	})
```

Comment:
Seems like this function is no longer needed - should call `TestAnalysisSingleTest` instead

Reply:
right. removed